Neos.Neos/Classes/Fusion/DimensionMenuItem.php

```diff
@@ -0,0 +1,25 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Neos\Neos\Fusion;
+
+use Neos\ContentRepository\Core\Projection\ContentGraph\Node;
+
+/**
+ * A menu item
+ */
+final class DimensionMenuItem
+{
+    /**
+     * @param array<string,mixed>|null $targetDimensions
+     */
+    public function __construct(
+        public readonly ?Node $node,
+        public readonly ?MenuItemState $state = null,
+        public readonly ?string $label = null,
+        public readonly ?array $targetDimensions = null,
+        public readonly ?string $uri = null
```

Comment:
nitpick: The doc comment could use some love and the class could be `readonly`
```suggestion
 * A menu item for dimension menus
 * Compared to the default {@see MenuItem} it has no `menuLevel` property, but one for the `targetDimensions`
 */
final readonly class DimensionMenuItem
{
    /**
     * @param array<string,mixed>|null $targetDimensions
     */
    public function __construct(
        public ?Node $node,
        public ?MenuItemState $state = null,
        public ?string $label = null,
        public ?array $targetDimensions = null,
        public ?string $uri = null
```

also, could we add some doc comments about the potential values of `$targetDimensions` – is that really an array or could we use one of the existing VOs?